Utilities/XrdAdaptor/src/QualityMetric.cc

```diff
@@ -157,16 +157,17 @@ QualityMetricFactory * QualityMetricFactory::m_instance = new QualityMetricFacto
 std::unique_ptr<QualityMetricSource>
 QualityMetricFactory::get(timespec now, const std::string &id)
```

Comment:
As far as I can tell, there is only ever one call to this function and it always get the value of "(unknown)"

http://cmslxr.fnal.gov/source/Utilities/XrdAdaptor/src/XrdSource.cc?v=CMSSW_8_0_X_2016-01-26-0000#0084

With that in mind, this code has a race condition when two requests with the same name come in.